querydsl-jpa/src/main/java/com/querydsl/jpa/JPAExpressions.java

```diff
@@ -102,6 +110,31 @@
         return select(expr).from(expr);
     }
 
+    /**
+     * Create a JPA 2.1 treated path.
+     *
+     * @param path The path to apply the treat operation on
+     * @param subtype subtype class
+     * @param <U> the subtype class
+     * @param <T> the expression type
+     * @return subtype instance with the same identity
+     */
+    public static <U extends BeanPath<? extends T>, T> U treat(BeanPath<? extends T> path, Class<U> subtype)  {
```

Comment:
@johnktims Apprently this is now considered a breaking change:

```
Class com.querydsl.jpa.JPAExpressions
 Added Method treat, sig <U:Lcom/querydsl/core/types/dsl/BeanPath<+TT;>;T:Ljava/lang/Object;>(Lcom/querydsl/core/types/dsl/BeanPath<+TT;>;Ljava/lang/Class<TU;>;)TU;, desc (Lcom/querydsl/core/types/dsl/BeanPath;Ljava/lang/Class;)Lcom/querydsl/core/types/dsl/BeanPath;, access public static
[WARNING] Rule 0: org.semver.enforcer.RequireBackwardCompatibility failed with message:
Current codebase is not backward compatible (BACKWARD_COMPATIBLE_USER) with version <4.3.2-20200601.233131-46>. Compatibility type has been detected as <NON_BACKWARD_COMPATIBLE>
```

How should we resolve this? Bump the snapshot version? Move the method to a new class? Add some sort of surpression?